packaging/os/slackpkg.py

```diff
@@ -0,0 +1,183 @@
+
+    import glob
+    import platform
+
+    machine = platform.machine()
+    packages = glob.glob("/var/log/packages/%s-*-[%s|noarch]*" % (name, machine))
+
+    if len(packages) > 0:
+        return True
+
+    return False
+
+
+def remove_packages(module, slackpkg_path, packages):
+    
+    remove_c = 0
+    # Using a for loop incase of error, we can report the package that failed
+    for package in packages:
+        # Query the package first, to see if we even need to remove
+        if not query_package(module, slackpkg_path, package):
+            continue
+
+        if not module.check_mode:
+            rc, out, err = module.run_command("%s -default_answer=y -batch=on remove %s" % (slackpkg_path, package))
+
+        if not module.check_mode and query_package(module, slackpkg_path, package):
+            module.fail_json(msg="failed to remove %s: %s" % (package, out))
+    
+        remove_c += 1
+
+    if remove_c > 0:
+
+        module.exit_json(changed=True, msg="removed %s package(s)" % remove_c)
+
+    module.exit_json(changed=False, msg="package(s) already absent")
+
+
+def install_packages(module, slackpkg_path, packages):
+
+    install_c = 0
+
+    for package in packages:
+        if query_package(module, slackpkg_path, package):
+            continue
+
+        if not module.check_mode:
+            rc, out, err = module.run_command("%s -default_answer=y -batch=on install %s" % (slackpkg_path, package))
+
+        if not module.check_mode and not query_package(module, slackpkg_path, package):
+            module.fail_json(msg="failed to install %s: %s" % (package, out), stderr=err)
+
+        install_c += 1
+    
+    if install_c > 0:
+        module.exit_json(changed=True, msg="present %s package(s)" % (install_c))
+
+    module.exit_json(changed=False, msg="package(s) already present")
+
+def upgrade_packages(module, slackpkg_path, packages):
+
+    install_c = 0
+
+    for package in packages:
+        if not module.check_mode:
+            rc, out, err = module.run_command("%s -default_answer=y -batch=on upgrade %s" % (slackpkg_path, package))
+
+        if not module.check_mode and not query_package(module, slackpkg_path, package):
+            module.fail_json(msg="failed to install %s: %s" % (package, out), stderr=err)
+
+        install_c += 1
+    
+    if install_c > 0:
+        module.exit_json(changed=True, msg="present %s package(s)" % (install_c))
+
+    module.exit_json(changed=False, msg="package(s) already present")
+
+def update_cache(module, slackpkg_path):
+    rc, out, err = module.run_command("%s -batch=on update" % (slackpkg_path))
+    if rc != 0:
+        module.fail_json(msg="Could not update package cache")
+
+def main():
+    module = AnsibleModule(
+        argument_spec       = dict(
+            state           = dict(default="installed", choices=['installed', 'removed', 'absent', 'present', 'latest']),
+            name            = dict(aliases=["pkg"], required=True),
```

Comment:
use `type=list` and you can pass a comma separated string as well as a yaml list. The result would be a python list, so your code must be adjusted accordantly